pkg/subscription/subscriptionmanager.go

```diff
@@ -155,94 +136,57 @@ func (sm *Manager) GetSubscriptionSchemaName(pd config.PolicyDetail) string {
 	return ""
 }
 
+// ProcessUnsubscribe moves a subscription from Approved to Active.
 func (sm *Manager) ProcessSubscribe(subscription apic.Subscription) {
-	log := sm.log.
-		WithField("subscriptionName", subscription.GetName()).
-		WithField("subscriptionID", subscription.GetID()).
-		WithField("catalogItemID", subscription.GetCatalogItemID()).
-		WithField("remoteID", subscription.GetRemoteAPIID()).
-		WithField("consumerInstanceID", subscription.GetApicID())
-
-	if err := sm.processSubscribe(subscription, log); err != nil {
-		log.WithError(err)
+	log := sm.log.WithFields(logFields(subscription))
+	if err := sm.processForState(subscription, log, apic.SubscriptionApproved); err != nil {
+		log.WithError(err).Error("failed to update subscription state")
 	}
 }
 
-func (sm *Manager) processSubscribe(subscription apic.Subscription, log logrus.FieldLogger) error {
+// processForState processes a subscription state change based on the current state.
+func (sm *Manager) processForState(subscription apic.Subscription, log logrus.FieldLogger, state apic.SubscriptionState) error {
```

Comment:
ProcessSubscribe and ProcessUnsubscribe were mostly the same except for the underlying function they called on the policy, which was subscribe or unsubscribe. They have been combined into one function and determine which one to call on the subscription based on the `state` parameter.